cmd/prune_branches.go

```diff
@@ -0,0 +1,73 @@
+package cmd
+
+import (
+	"log"
+
+	"github.com/Originate/git-town/lib/config"
+	"github.com/Originate/git-town/lib/git"
+	"github.com/Originate/git-town/lib/script"
+	"github.com/Originate/git-town/lib/steps"
+	"github.com/spf13/cobra"
+)
+
+type PruneBranchesFlags struct {
+	Undo bool
+}
+
+var pruneBranchesFlags PruneBranchesFlags
+
+var pruneBranchesCommand = &cobra.Command{
+	Use:   "prune-branches",
+	Short: "Deletes local branches whose tracking branch no longer exists",
+	Long:  `Deletes local branches whose tracking branch no longer existsh`,
+	Run: func(cmd *cobra.Command, args []string) {
+		steps.Run(steps.RunOptions{
+			CanSkip:              func() bool { return false },
+			Command:              "prune-branches",
+			IsAbort:              false,
+			IsContinue:           false,
+			IsSkip:               false,
+			IsUndo:               pruneBranchesFlags.Undo,
+			SkipMessageGenerator: func() string { return "" },
+			StepListGenerator: func() steps.StepList {
+				checkPruneBranchesPreconditions()
+				return getPruneBranchesList()
+			},
+		})
+	},
+}
+
+func checkPruneBranchesPreconditions() {
+	if config.HasRemote("origin") {
+		fetchErr := script.RunCommand("git", "fetch", "--prune")
```

Comment:
What do you think about extracting a `FetchRemoteCommits` method to make this more readable?